modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1398,8 +1398,7 @@ void GDScriptAnalyzer::resolve_parameter(GDScriptParser::ParameterNode *p_parame
 	}
 
 	if (p_parameter->datatype_specifier != nullptr) {
-		resolve_datatype(p_parameter->datatype_specifier);
-		result = p_parameter->datatype_specifier->get_datatype();
+		result = resolve_datatype(p_parameter->datatype_specifier);
```

Comment:
Technically this should be the same thing, but there seems to be some branches that don't set the datatype in the specifier (which is sort of a bug). I guess this format is clearer anyway.